File: examples/typescript/01_simple_click.ts
/**
 * Example 01: Simple Click
 *
 * This example demonstrates the most basic usage of the Computer Use Agent:
 * - Initialize the agent
 * - Give it a simple goal
 * - Let it analyze the screen and click on an element
 *
 * Use case: Opening an application or clicking a button
 */

import { ComputerUseAgent } from "../../typescript/src";

const apiKey = process.env.ANTHROPIC_API_KEY;
if (!apiKey) {
  console.error("❌ Please set ANTHROPIC_API_KEY environment variable");
  console.error("   export ANTHROPIC_API_KEY='your-key-here'");
  process.exit(1);
}

console.log("🤖 Computer Use Agent - Simple Click Example");
console.log("=".repeat(60));

async function main() {
  // Initialize the agent
  const agent = new ComputerUseAgent({
    apiKey,
    model: "claude-sonnet-4-20250514",
  });

  // Give the agent a simple goal
  const goal = "Find and click on the Chrome browser icon to open it";

  console.log(`\n🎯 Goal: ${goal}`);
  console.log(`\n⏳ Running agent (max 5 iterations)...\n`);

  // Run the agent
  const result = await agent.run(goal, 5);

  // Print results
  console.log("\n" + "=".repeat(60));
  console.log("📊 RESULTS");
  console.log("=".repeat(60));
  console.log(`✓ Success: ${result.success}`);
  console.log(`✓ Iterations used: ${result.iterations}`);
  console.log(`✓ Actions taken: ${result.actionHistory?.length || 0}`);

  if (!result.success) {
    console.log(`❌ Error: ${result.error}`);
  }

  console.log("\n💡 Tip: Try changing the goal to click on different elements!");
}

main().catch(console.error);

File: examples/python/01_simple_click.py
"""
Example 01: Simple Click

This example demonstrates the most basic usage of the Computer Use Agent:
- Initialize the agent
- Give it a simple goal
- Let it analyze the screen and click on an element

Use case: Opening an application or clicking a button
"""

from computer_use_agent import ComputerUseAgent
import os

# Get API key from environment
api_key = os.getenv("ANTHROPIC_API_KEY")
if not api_key:
    print("❌ Please set ANTHROPIC_API_KEY environment variable")
    print("   export ANTHROPIC_API_KEY='your-key-here'")
    exit(1)

print("🤖 Computer Use Agent - Simple Click Example")
print("=" * 60)

# Initialize the agent
agent = ComputerUseAgent(
    api_key=api_key,
    model="claude-sonnet-4-20250514",
)

# Give the agent a simple goal
goal = "Find and click on the Chrome browser icon to open it"

print(f"\n🎯 Goal: {goal}")
print(f"\n⏳ Running agent (max 5 iterations)...\n")

# Run the agent
result = agent.run(
    goal=goal,
    max_iterations=5,
)

# Print results
print("\n" + "=" * 60)
print("📊 RESULTS")
print("=" * 60)
print(f"✓ Success: {result['success']}")
print(f"✓ Iterations used: {result['iterations']}")
print(f"✓ Actions taken: {len(result.get('action_history', []))}")

if not result['success']:
    print(f"❌ Error: {result.get('error')}")

print("\n💡 Tip: Try changing the goal to click on different elements!")

